File: csharp/src/CQRSService.Client/Events/QueueEvents.cs
using CQRSService.Client.Domain.Events;
using CQRSService.Domain.Events;
using System;
using System.Threading;
using System.Threading.Tasks;

namespace CQRSService.Client.Events
{
    public class QueueEvents : 
        IEventHandler<ClientCreated>
        , IEventHandler<ClientRemoved>
        , IEventHandler<ClientUpdated>
    {
        public Task Handle(ClientCreated notification, CancellationToken cancellationToken)
        {
            return Task.CompletedTask;
        }

        public Task Handle(ClientRemoved notification, CancellationToken cancellationToken)
        {
            throw new NotImplementedException();
        }

        public Task Handle(ClientUpdated notification, CancellationToken cancellationToken)
        {
            throw new NotImplementedException();
        }
    }
}

File: golang/client/events/queueEvents.go
package client

import (
	"fmt"

	eventDomain "github.com/cqrs-govscsharp/golang/domain/events"
)

type QueueEvents struct {
	eventDomain.EventHadler
}

func NewQueueEvents() QueueEvents {
	return QueueEvents{}
}

func (eh QueueEvents) Handle(event eventDomain.Eventer) error {
	fmt.Printf("QueueEvents %s", event.Name())
	return nil
}

func (eh QueueEvents) EventNames() []string {
	return []string{"ClientCreated", "ClientRemoved", "ClientUpdated"}
}
